app/controllers/donations_controller.rb

```diff
@@ -13,9 +13,11 @@ def index
                                      .order(created_at: :desc)
                                      .where(issued_at: date_range)
                                      .class_filter(filter_params)
+    @paginated_donations = @donations.page(params[:page])
```

Comment:
I *think* if you do:
```suggestion
                                     .page(params[:page])
```
it would just apply the `limit` to the initial query, which would reduce the resultset size. (Pls verify if you do this; I've not used Kaminari in a couple years)